lib/execution_engine2/db/MongoUtil.py

```diff
@@ -223,6 +223,48 @@ def get_job(self, job_id=None, exclude_fields=None) -> Job:
 
         return job
 
+    def get_jobs_with_status(
+        self,
+        job_ids: List[str],
+        status_list: List[str],
+        only_job_ids: bool = False,
+        retried_jobs_allowed=True,
+    ) -> Union[List[Job], List[str]]:
+        if not (job_ids and isinstance(job_ids, list)):
+            raise ValueError("Please provide a non empty list of job ids")
+
+        if not (status_list and isinstance(job_ids, list)):
```

Comment:
should that second one be `isinstance(status_list, list)`?

It would be good to put in some tests for this so you can check what is and isn't caught by these conditionals vs by the type checking in the function definition. It's sometimes easier to have arguments that should be a list default to `None` so that you don't have to check the length of the list.